psyneulink/core/compositions/composition.py

```diff
@@ -6597,13 +6643,17 @@ def run(
             assert not is_simulation
             try:
                 if bin_execute is True or bin_execute.startswith('LLVM'):
-                    _comp_ex = pnlvm.CompExecution(self, [context.execution_id])
+                    comp_ex_tags = []
+                    # Add learning tags if the composition is to be ran in learning mode
+                    from psyneulink.library.compositions.autodiffcomposition import AutodiffComposition
+                    if self._is_learning(context) and isinstance(self, AutodiffComposition):
```

Comment:
Why does this need to check for autodiff instance? `isinstance` calls are rather expensive.
Use set for tags instead of list.
This looks like it will break CUDA execution.

Reply:
I wasn't sure if adding `learning` to tags would face an error in other compiled run situations, so I made it only add it in the case of autodiff

Can you elaborate on how it would break CUDA, and how I could possibly fix that? I can't test CUDA on my machine, so it would be good to know.

Reply:
There shouldn't be any compiled learning tests other than autodiff, so ti should be safe to drop. CUDA should match what LLVM. In this case adding the new tag in __ptx_initialize should be enough. you can ping me to test CUDA execution.